cmake/CeleritasUtils.cmake

```diff
@@ -73,13 +85,66 @@ CMake utility functions for Celeritas.
     [<INTERFACE|PUBLIC|PRIVATE> [items2...] ...])
 
 #]=======================================================================]
-include(FindPackageHandleStandardArgs)
+include_guard(GLOBAL)
 
-macro(celeritas_find_package_config _package)
-  find_package(${_package} CONFIG ${ARGN})
-  find_package_handle_standard_args(${_package} CONFIG_MODE)
+include(CheckLanguage)
+
+#-----------------------------------------------------------------------------#
+# CONFIGURATION UTILITIES
+#-----------------------------------------------------------------------------#
+
+function(celeritas_optional_language lang)
+  set(_var "CELERITAS_USE_${lang}")
+  if(DEFINED "${_var}")
+    set(_val "${_var}")
+  else()
+    check_language(${lang})
+    set(_val OFF)
+    if(CMAKE_${lang}_COMPILER)
+      set(_val ON)
+    endif()
+    message(STATUS "Set ${_var}=${_val} based on compiler availability")
+  endif()
+
+  option("${_var}" "Enable the ${lang} language" "${_val}" )
+endfunction()
+
+function(celeritas_to_onoff varname)
+  if(ARGC GREATER 1 AND ARGV1)
+    set(${varname} ON PARENT_SCOPE)
+  else()
+    set(${varname} OFF PARENT_SCOPE)
+  endif()
+endfunction()
+
+# Note: this is a macro so that `find_package` variables stay in the global
+# scope.
+macro(celeritas_optional_package package)
+  if("${ARGC}" EQUAL 2)
+    set(_findpkg "${package}")
+    set(_docstring "${ARGV1}")
+  else()
+    set(_findpkg "${ARGV1}")
+    set(_docstring "${ARGV2}")
+  endif()
+
+  set(_var "CELERITAS_USE_${package}")
+  if(DEFINED "${_var}")
+    set(_val "${_var}")
+  else()
+    find_package(${_findpkg} QUIET)
```

Comment:
If I read the code correctly, the call to `celeritas_optional_package` will execute a quiet `find_package` and then later on a non-quiet `find_package` is done with addition arguments (in particular the request version).  Doesn't the quiet find_package interfer with the non-quiet find_package in case there is multiple version of a package available? (I.e. the 2nd `find-package` might be ignored)

Reply:
Not as far as I can tell: the second `find_package` does all the requirements checks even if the package is already found earlier. Even if it somehow skips that check the first time, subsequent configures won't even do the first `find_package` so we should be able to tell if such a situation arises.